src/core_scan/Core_scan.ml

```diff
@@ -763,12 +769,13 @@ let extracted_targets_of_config (config : Core_scan_config.t)
              add_additional_targets config (List.length extracted_targets);
            extracted_targets)
   in
-  List.fold_right
-    (fun (t, fn) (ts, fn_tbl) ->
-      Hashtbl.add fn_tbl t.In.path fn;
-      (t :: ts, fn_tbl))
+  let num_targets = List.length basic_targets in
+  (List.fold_right (fun (t, fn, (newf, origf)) (ts, fn_tbl, file_tbl) ->
+       Hashtbl.add fn_tbl t.Input_to_core_t.path fn;
+       Hashtbl.add file_tbl newf origf;
+       (t :: ts, fn_tbl, file_tbl)))
     extracted_ranges
-    ([], Hashtbl.create (List.length basic_targets))
+    ([], Hashtbl.create num_targets, Hashtbl.create num_targets)
```

Comment:
Hash tables being mutable, let's not pass them around. A mechanical transformation gives me something like this:
```ocaml
let fn_tbl = Hashtbl.create num_targets in
let file_tbl = Hashtbl.create num_targets in
let ??? =
  List.map (fun (t, fn, (newf, origf)) ts ->
    Hashtbl.add fn_tbl t.Input_to_core_t.path fn;
    Hashtbl.add file_tbl newf origf;
    t :: ts)
  extracted_ranges
in
(???, fn_tbl, file_tbl)
```

Reply:
or do a `List.iter` pass to populate the tables and a separate `List.map` to build the list.

Reply:
In general, it's easier to keep concerns separate.

Reply:
I was planning to follow the paradigm of the original code, but you're right, I really don't like this